var/spack/repos/builtin/packages/wgrib2/package.py

```diff
@@ -2,20 +2,65 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-import os
 import re
 
+import spack.build_systems.cmake
+import spack.build_systems.makefile
 from spack.package import *
 
-
-class Wgrib2(MakefilePackage):
+variant_map_common = {
+    "netcdf3": "USE_NETCDF3",
+    "netcdf4": "USE_NETCDF4",
+    "mysql": "USE_MYSQL",
+    "udf": "USE_UDF",
+    "regex": "USE_REGEX",
+    "tigge": "USE_TIGGE",
+    "proj4": "USE_PROJ4",
+    "aec": "USE_AEC",
+    "g2c": "USE_G2CLIB",
+    "png": "USE_PNG",
+    "jasper": "USE_JASPER",
+    "openmp": "USE_OPENMP",
+    "wmo_validation": "USE_WMO_VALIDATION",
+    "ipolates": "USE_IPOLATES",
+    "disable_alarm": "DISABLE_ALARM",
+    "fortran_api": "MAKE_FTN_API",
+    "disable_stat": "DISABLE_STAT",
+    "openjpeg": "USE_OPENJPEG",
+}
+
+variant_map_makefile = {"spectral": "USE_SPECTRAL"}
+
+variant_map_cmake = {
+    "netcdf": "USE_NETCDF",
+    "disable_timezone": "DISABLE_TIMEZONE",
+    "enable_docs": "ENABLE_DOCS",
+}
+
+variant_map_makefile.update(variant_map_common)
+variant_map_cmake.update(variant_map_common)
+
+
+class Wgrib2(MakefilePackage, CMakePackage):
     """Utility for interacting with GRIB2 files"""
 
     homepage = "https://www.cpc.ncep.noaa.gov/products/wesley/wgrib2"
     url = "https://www.ftp.cpc.ncep.noaa.gov/wd51we/wgrib2/wgrib2.tgz.v2.0.8"
+    git = "https://github.com/NOAA-EMC/wgrib2"
```

Comment:
I'd update the `url` here too to pick up newer versions correct with `spack checksum` (also fix the git url to end with `.git`).
```suggestion
    url = "https://github.com/NOAA-EMC/wgrib2/archive/refs/tags/v3.5.0.tar.gz"
    git = "https://github.com/NOAA-EMC/wgrib2.git"
```

Reply:
Makes sense re: url, thanks. I'm curious what's the benefit of the '.git'?

Reply:
I think without the git it is just a webpage. Might be that github is actually smart enough about it, but it used to give trouble.